### PR TITLE
Add more documentation

### PR DESCRIPTION
**Pull request by alteous**
This PR intends to document the whole project more thoroughly.

### CODE REVIEW

**Comment by alteous** [comment]
I'm not finished with this yet. The PR is for anyone to submit constructive criticism and/or request documentation in lacking areas.

File: examples/tutorial.rs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    let center = [0.0, 0.0];
+    let yextent = 1.0;
+    let zrange = -1.0 .. 1.0;
+    let camera = window.factory.orthographic_camera(center, yextent, zrange);
+    while window.update() {
+        window.render(&camera);
+    }
+}
+
+//! The same code but in doc comment format.
```
Comment (kvark) [inline examples/tutorial.rs]: mm, duplicating code is quite unfortunate. Can we use some `include!` instead?

Comment (alteous) [reply examples/tutorial.rs]: I've removed this in the latest commit.

**Comment by alteous** [comment]
I'm ready to merge this if there are no further comments.

**Comment by alteous** [comment]
Sure.

**Pull request merged**